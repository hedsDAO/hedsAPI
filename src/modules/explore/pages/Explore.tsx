export const Explore = ({ wallet }: { wallet?: string }) => {
  return <></>;
};
