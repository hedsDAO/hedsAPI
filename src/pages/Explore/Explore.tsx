import React from 'react';
import { doc, getDoc } from 'firebase/firestore';
import { db } from '../../App';
import { useSignMessage } from 'wagmi';

export const Explore = ({ wallet }: { wallet?: string }) => {
  const { data, isError, isLoading, isSuccess, signMessage } = useSignMessage({
    message: 'heds heds heds',
  });
  console.log(data);
  return (
    <>
      {' '}
      <div className='mx-auto max-w-7xl text-center'>
        <button className="bg-black text-sm rounded-full text-white px-3 py-1" disabled={isLoading} onClick={() => signMessage()}>
          Sign message
        </button>
        {isSuccess && <div>Signature: {data}</div>}
        {isError && <div>Error signing message</div>}
      </div>
    </>
  );
};
