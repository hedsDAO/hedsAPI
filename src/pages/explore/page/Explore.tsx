import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Dispatch, store } from '@/store';
import { Box, Divider, Image } from '@chakra-ui/react';
import { LatestRelease, CarouselWrapper, HedsSolo, MostAppearances, HedsPlayer, Stats } from '@/pages/explore/components';
import { HEDS_IMG_PROPS, MAX_TOP_ARTISTS } from '@/pages/explore/store/constants';
import hedImage from '../../../public/heddot.png';

export const Explore = () => {
  const dispatch = useDispatch<Dispatch>();
  const allArtists = useSelector(store.select.artistModel.selectAllArtists);
  useEffect(() => {
    if (allArtists) dispatch.artistModel.getMostFeaturedArtists([allArtists, MAX_TOP_ARTISTS]);
  }, [allArtists]);

  useEffect(() => {
    dispatch.exploreModel.getLatestSecondaryListings();
  }, []);

  return (
    <Box>
      <CarouselWrapper slides={[<LatestRelease />, <HedsSolo />, <HedsPlayer />]} />
      <Image src={hedImage} {...HEDS_IMG_PROPS} />
      <MostAppearances />
      <Divider mx="auto" w="80%" my={5} />
      <Stats />
      <Divider mx="auto" w="80%" my={5} />
      {/* <ActiveListings /> */}
    </Box>
  );
};
