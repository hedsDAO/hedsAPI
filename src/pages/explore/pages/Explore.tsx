import { Dispatch, RootState } from '@/store';
import { Box, Container, Divider } from '@chakra-ui/react';
import { Fragment, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

export const Explore = ({ wallet }: { wallet?: string }) => {
  const dispatch = useDispatch<Dispatch>();
  useEffect(() => {
    dispatch.exploreModel.getExploreStats();
  }, []);
  const { stats } = useSelector((state: RootState) => state.exploreModel);
  return (
    <Fragment>
      {stats && (
        <Box display={{ base: 'none', lg: 'block' }} bgColor={'blackAlpha.900'} pt={{ base: '8', sm: '12' }}>
          <div className="bg-white">
            <div className="relative">
              <Box inset={0} h={'50%'} pos={'absolute'} bgColor={'blackAlpha.900'} />
              <div className="relative mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
                <div className="mx-auto max-w-2xl">
                  <dl className="rounded-lg bg-white shadow-lg sm:grid sm:grid-cols-3">
                    <div className="flex flex-col border-b border-gray-100 px-2 py-3 text-center sm:border-0 sm:border-r">
                      <dt className="order-2 mt-2 text-sm font-medium leading-6 text-gray-500">Value Generated</dt>
                      <dd className="order-1 text-2xl font-bold tracking-tight text-neutral-800">
                        {/* {formatDollars(stats.earnings.primary.usd + stats.earnings.secondary.usd)} */}
                      </dd>
                    </div>
                    <div className="flex flex-col border-t border-b border-gray-100 px-2 py-3 text-center sm:border-0 sm:border-l sm:border-r">
                      <dt className="order-2 mt-2 text-sm font-medium leading-6 text-gray-500">Total Submissions</dt>
                      <dd className="order-1 text-2xl font-bold tracking-tight text-neutral-800">{stats.submissions.total}</dd>
                    </div>
                    <div className="flex flex-col border-t border-gray-100 px-2 py-3 text-center sm:border-0 sm:border-l">
                      <dt className="order-2 mt-2 text-sm font-medium leading-6 text-gray-500">Tapes Minted</dt>
                      <dd className="order-1 text-2xl font-bold tracking-tight text-neutral-800">{stats.tapes.total}</dd>
                    </div>
                  </dl>
                </div>
              </div>
            </div>
          </div>
        </Box>
      )}
      {/* {latestTape && (
        <Container maxW="7xl">
          <div>
            <div className="mx-auto w-full px-4 lg:px-1">
              <Heading fontWeight={'bold'} letterSpacing={'tight'} size={['lg', 'xl']} color={'gray.800'} mt={{ base: 5, lg: 10 }} mb={3} p={1}>
                Live Tapes
              </Heading>
            </div>
            <dl className="mt-5 grid grid-cols-1 gap-5 sm:grid-cols-2 lg:grid-cols-3">
              <div className="relative overflow-hidden rounded-lg bg-white px-3 shadow pb-12 lg:px-6">
                <div className="font-semibold text-lg w-full py-4">{latestTape.name}</div>
                <dt>
                  <div className="absolute rounded-md">
                    <Avatar borderRadius={'md'} size="lg" className="aspect-square" src={latestTape.image} />
                  </div>
                </dt>
                <dd className="ml-20 flex items-baseline pb-6 sm:pb-7">
                  {timeline &&
                    Object.entries(timeline).map(([string, stepData]) => {
                      if (string === 'submit' && stepData.status === TimelineStatus.OPEN) return <Submit />;
                      if (string === 'vote' && stepData.status === TimelineStatus.OPEN) return <Vote />;
                      if (string === 'mint' && stepData.status === TimelineStatus.OPEN) return <Mint />;
                    })}
                  <div className="absolute inset-x-0 bottom-0 bg-gray-50 px-4 py-4 sm:px-6">
                    <div className="text-sm">
                      <a href="#" className="font-medium text-blue-400 hover:text-blue-400">
                        View tape
                      </a>
                    </div>
                  </div>
                </dd>
              </div>
            </dl>
          </div>
        </Container>
      )} */}
      <Container maxW={'7xl'}>
        <Divider mt={10} />
        {/* <TopArtists /> */}
      </Container>
    </Fragment>
  );
};
