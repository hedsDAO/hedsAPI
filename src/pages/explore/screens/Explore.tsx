import { useState } from 'react';
import { Box, Flex, Image, Center, useBreakpointValue } from '@chakra-ui/react';
import { motion, AnimatePresence } from 'framer-motion';

const MotionBox = motion(Box);

export const Explore = () => {
  const items = [
    'https://www.heds.cloud/ipfs/QmSy3SvxY9HPvH2cEjb1NT8Y5rfcCmaWEge2vxhFvRvXj1',
    'https://www.heds.cloud/ipfs/QmbnGKrUsfRwxcPNxdXuKwHFmtu29F7UYSAfY5aEMGZtPc',
    'https://www.heds.cloud/ipfs/QmY2z5HEK6rD3SQ27MPpitZmqeqywKEwB7Ci1ABjqwe6J9',
    'https://www.heds.cloud/ipfs/QmfDej1TKoo7Aqi1Pf5P6WGzPus4RzWi2tq77U3MzxzWt6',
    'https://www.heds.cloud/ipfs/QmQTv6yF5rQCxwgXa6AKHJZGx1NUwS4KVAwjrg6muLzufJ',
  ];
  const [activeIndex, setActiveIndex] = useState(2);
  const imageSize = useBreakpointValue({ base: '200px', md: '300px' });
  const imageSizeInteger = useBreakpointValue({ base: 200, md: 300 });
  const handlePrev = () => setActiveIndex((old) => (old === 0 ? items.length - 1 : old - 1));
  const handleNext = () => setActiveIndex((old) => (old === items.length - 1 ? 0 : old + 1));
  const calculateZIndex = (offset: number) => (offset === 0 ? 2 : offset === 1 || offset === -1 ? 1 : 0);
  const calculateScale = (offset: number) => (offset === 0 ? 1.3 : 1);

  return (
    <Center minH="400px" position="relative" style={{ perspective: 1000 }}>
      <MotionBox position="absolute" left={0} onClick={handlePrev} p={4} cursor="pointer" fontSize="2xl" zIndex="1" color="white">
        {'<'}
      </MotionBox>
      <Flex w="full" overflow="hidden" justifyContent="center" alignItems="center">
        <AnimatePresence initial={false} custom={activeIndex}>
          {items.map((src, i) => {
            const offset = i - activeIndex;
            const isCenter = offset === 0;
            return (
              <MotionBox
                key={i}
                position="absolute"
                shadow="md"
                zIndex={calculateZIndex(offset)} // Adjust zIndex based on position
                initial={{ x: offset * 200 }} // Adjust x value here
                animate={{
                  x: offset * 200,
                  scale: calculateScale(offset),
                  rotateY: isCenter ? 0 : '45deg',
                  // height: isCenter ? imageSize: '250px',
                  // width: isCenter ? imageSize: '300px',
                }}
                transition={{ type: 'spring', damping: 20, stiffness: 100 }}
              >
                <Image onClick={() => setActiveIndex(i)} rounded={'lg'} src={src} h={imageSize} w={imageSize} />
              </MotionBox>
            );
          })}
        </AnimatePresence>
      </Flex>
      <MotionBox position="absolute" right={0} onClick={handleNext} p={4} cursor="pointer" fontSize="2xl" zIndex="1" color="white">
        {'>'}
      </MotionBox>
    </Center>
  );
};
