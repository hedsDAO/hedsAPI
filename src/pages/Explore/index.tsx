import React from 'react';
import { Explore } from './Explore';
export default (): JSX.Element => <Explore />;
