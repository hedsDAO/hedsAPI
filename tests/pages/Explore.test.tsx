import React from 'react';
import { Explore } from '@pages/Explore/Explore';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

describe('Explore Unit', () => {
  test('should render component properly', () => {
    const render = renderer.create(<Explore />);
    const tree = render.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
