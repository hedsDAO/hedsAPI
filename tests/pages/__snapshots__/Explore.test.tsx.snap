// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore Unit should render component properly 1`] = `"explore"`;
